src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewSparkConnectionDialog.java

```diff
@@ -273,9 +273,17 @@ public void execute()
                builder.append("library(dplyr)\n");
             }
             
+            String[] masterComponents = master_.getSelection().split(" - ");
+
             builder.append("sc <- spark_connect(master = \"");
-            builder.append(master_.getSelection());
+            builder.append(masterComponents[0]);
             builder.append("\"");
+            if (masterComponents.length > 0)
```

Comment:
Yes! In practice, all the date entries are a "master - appname" pair, but added this to be resilient in case this changes in the future, thanks for pointing this out.